src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -474,16 +471,18 @@ class NetworkOPsImp final
     void onStop () override
     {
         mAcquiringLedger.reset();
-        m_heartbeatTimer.cancel();
-        m_clusterTimer.cancel();
+
+        if (heartbeatTimer_)
+            heartbeatTimer_->cancel();
+        if (clusterTimer_)
+            clusterTimer_->cancel();
 
         stopped ();
```

Comment:
Excellent point.  What's the correct method for determining whether a completion handler is running?  I did not see a technique when I looked at `asio::steady_timer`, but I probably missed it.